### PR TITLE
Fix bare imports in AssetFinder subdirectories

### PR DESCRIPTION
**Pull request by dbnicholson**
When creating a non-root AssetFinder, copy the parent's `extract_packages` so that importing bare modules works correctly. This is typically needed when packages are vendored and `sys.path` is dynamically updated to prefer the vendored version.

Fixes: #820

### CODE REVIEW

**Comment by dbnicholson** [comment]
This seems to be all that's needed. If the added test is run without copying the parent's `extract_packages`, it fails in the same way as described in #820. I also tested this in my application and it worked correctly. I don't know if there are any side effects from the change, though.

**Comment by mhsmith** [comment]
Thanks, I'll have a look at this and your other PR in a few days.

File: product/runtime/src/test/python/chaquopy/test/android/test_import.py [hunk 1]
```diff
@@ -581,18 +581,18 @@ def check_top_level(real_name, load_name, id):
         # important enough to investigate just now.
         self.assertFalse(hasattr(imp_rename_2, "mod_3"))
 
-    # Ensure that a package can be imported by a bare name when its in an AssetFinder subdirectory.
+    # Ensure that a package can be imported by a bare name when it's in an AssetFinder subdirectory.
     # This is typically done when vendoring packages and dynamically adjusting sys.path. See #820.
-    def test_imp_subdir(self):
-        sys.modules.pop("about", None)
+    def test_path_subdir(self):
```
Comment (mhsmith) [inline product/runtime/src/test/python/chaquopy/test/android/test_import.py:586]: "imp" in the other tests refers to the `imp` module, which this isn't related to.

**Review by mhsmith — COMMENTED** [review]
Thanks very much for this, and sorry for taking so long to review it. The implementation change looks fine, but there was a problem with the test – it didn't reproduce the original problem, because `murmurhash.about` is a module, not a package. So I've changed it to use `murmurhash.tests`.

**Pull request merged**